src/current/v23.2/protect-changefeed-data.md

```diff
@@ -0,0 +1,79 @@
+
+However, if the changefeed lags too far behind, the protected changes could lead to an accumulation of garbage. This could result in increased disk usage and degraded performance for some workloads.
+
+## Prevent garbage accumulation
+
+To prevent an accumulation of protected changes that could impact performance, consider the following:
+
+- [`changefeed.protect_timestamp.max_age`](#changefeed-protect_timestamp-max_age)
+- [`gc_protect_expires_after`](#gc_protect_expires_after)
+
+### `changefeed.protect_timestamp.max_age`
+
+{% include_cached new-in.html version="v23.2" %} By default, the `changefeed.protect_timestamp.max_age` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) sets the maximum time that changefeeds making no forward progress will hold protected timestamp records. Once the `changefeed.protect_timestamp.max_age` duration is reached, the changefeed will fail with a permanent error. As a result, it is **critical to monitor for changefeed failures** because changefeeds will eventually fail with an unrecoverable error if they cannot progress before the duration is reached.
+
+This cluster setting is enabled by default to 4 days.
+
+`changefeed.protect_timestamp.max_age` is a cluster-wide setting affecting all changefeeds.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING changefeed.protect_timestamp.max_age = '120h';
+~~~
+
+{{site.data.alerts.callout_info}}
+`changefeed.protect_timestamp.max_age` applies only to **newly created changefeeds in v23.2**.
+
+If you are [upgrading to v23.2]({% link {{ page.version.version }}/upgrade-cockroach-version.md %}), it is necessary to set `protect_data_from_gc_on_pause` on any existing changefeeds to ensure that it does not enter a situation of infinite retries, which could prevent garbage collection. You can use the [`ALTER CHANGEFEED`]({% link {{ page.version.version }}/alter-changefeed.md %}) statement to add `protect_data_from_gc_on_pause` to existing changefeeds.
+{{site.data.alerts.end}}
+
+### `gc_protect_expires_after`
+
+The [`gc_protect_expires_after`]({% link {{ page.version.version }}/create-changefeed.md %}#gc-protect-expire) option automatically expires the protected timestamp records that are older than the defined duration and cancels a changefeed job.
+
+For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE db.table INTO 'external://sink' WITH on_error='pause', gc_protect_expires_after='24h';
+~~~
+
+If this changefeed runs into a retryable error, protected timestamps will protect changes for up to 24 hours. After this point, if the changefeed has not resumed, the protected timestamp records will expire and the changefeed job will be canceled to prevent accumulation of garbage.
```

Comment:
"has not resumed" probably needs to be changed (changefeed does not need to be resumed -- it's probably running, and is retry once in a while to try to deliver a message)
Something like:
... If the changefeed has not made any progress in the past 24 hours, the protected ....

Reply:
Done!